src/centralized_telescope/wrapper.rs

```diff
@@ -0,0 +1,38 @@
+use super::algorithm;
+use super::init::make_setup;
+use super::params::Params;
+use super::proof::Proof;
+use super::setup::Setup;
+use crate::utils::types::Element;
+
+/// The main ALBA struct with prove and verify functions.
+#[derive(Debug, Clone, Copy)]
+pub struct Wrapper {
+    setup: Setup,
+}
+
+impl Wrapper {
+    /// Initialize ALBA with `Params`.
+    pub fn create(params: &Params) -> Self {
+        let setup = make_setup(params);
+        Self::create_with_setup(&setup)
+    }
+
+    /// Initialize ALBA with `Setup`. This function is unsafe to use and should
+    /// be avoided.
+    pub fn create_with_setup(setup: &Setup) -> Self {
+        Self { setup: *setup }
+    }
+
+    /// Alba's proving algorithm, based on a depth-first search algorithm.
+    /// Returns either a `Proof` or `None` if no proof is found.
+    pub fn prove(&self, prover_set: &[Element]) -> Option<Proof> {
+        algorithm::prove(&self.setup, prover_set)
+    }
+
+    /// Alba's verification algorithm.
+    /// Returns true if and only if the proof is successfully verified.
+    pub fn verify(&self, proof: &Proof) -> bool {
+        algorithm::verify(&self.setup, proof)
+    }
+}
```

Comment:
The tests should be moved to the user facing wrapper

Reply:
I agree that tests should be moved. Let's do this in a separate PR?

Reply:
#79